### DISCUSSION TITLE
Error in api request in python

### DISCUSSION THREAD

**Discussion by MalikTayyabTanveer** [deepgram/community#996] [General help] [answered]
i am using the deepgram-sdk==2.12.0
async def get_transcript(callback):
    dg_client = Deepgram(os.getenv("DEEPGRAM_API_KEY"))
    transcription_complete = asyncio.Event()  # Event to signal transcription completion
    try:
        # Define options for live transcription
        options = {
            "punctuate": True,
            "model": "nova-2",
            "language": "en-US",
            "encoding": "linear16",
            "channels": 1,
            "sample_rate": 16000,
            "endpointing": 300,
            "smart_format": True,
        }

        # Message handling function for received transcript
        async def on_transcript_received(data):
            if "channel" in data and "alternatives" in data["channel"]:
                sentence = data["channel"]["alternatives"][0].get("transcript", "")

                if not data.get("is_final", False):
                    transcript_collector.add_part(sentence)
                else:
                    # Final sentence part
                    transcript_collector.add_part(sentence)
                    full_sentence = transcript_collector.get_full_transcript()

                    if full_sentence.strip():
                        print(f"Human: {full_sentence.strip()}")
                        callback(full_sentence.strip())
                        transcript_collector.reset()
                        transcription_complete.set()

        # Start live transcription
        print("Listening...")
        connection = await dg_client.transcription.live(options=options, callback=on_transcript_received)

        await transcription_complete.wait()  # Wait for transcription to complete
        await connection.close()  # Close the connection

    except Exception as e:
        print(f"Could not open socket: {e}")
        
this is my code. but getting the error: Could not open socket: server rejected WebSocket connection: HTTP 400
i do not know why. since the maintenance of the deepgram, i am getting this error. can anyone tell me about this

**Comment by deepgram-community** [comment]
Hey there! It looks like you haven't connected your GitHub account to your Deepgram account. You can do this at https://community.deepgram.com - being verified through this process will allow our team to help you in a much more streamlined fashion.

**Comment by deepgram-community** [comment]
It looks like we're missing some important information to help debug your issue. Would you mind providing us with the following details in a reply?

- The deepgram product you are using (e.g Speech to Text, Agent API)
- A request ID that triggered your error or issue.

**Comment by deepgram-community** [comment, marked as answer]
our Python SDK 2.12 version is very old we recommend using the latest version of our Python SDK as we can't guarantee backwards compatibility on older SDK versions.

Can you try running a new version and see if this resolves the problem?

<https://github.com/deepgram/deepgram-python-sdk/releases/tag/v3.7.6>

> ![avatar](https://avatars.slack-edge.com/2023-02-20/4855338233600_fad9174023ee3f640ce1_32.png) This message was sent by John Vajda from Deepgram, via our community automation.
